EIPS/eip-7612.md

```diff
@@ -0,0 +1,181 @@
+
+```python3
+@dataclass
+class State:
+    """
+    Contains all information that is preserved between transactions.
+    """
+
+    _main_trie: Trie[Address, Optional[Account]] = field(
+        default_factory=lambda: Trie(secured=True, default=None)
+    )
+    _storage_tries: Dict[Address, Trie[Bytes, U256]] = field(
+        default_factory=dict
+    )
+    _snapshots: List[
+        Tuple[
+            Trie[Address, Optional[Account]], Dict[Address, Trie[Bytes, U256]]
+        ]
+    ] = field(default_factory=list)
+    _created_accounts: Set[Address] = field(default_factory=set)
+
+    # Added in this EIP
+    _overlay_tree: VerkleTree[Address, Bytes32]
+```
+
+And the state access functions are modified as such:
+
+```python3
+def get_account_optional(state: State, address: Address) -> Optional[Account]:
+    account = verkle_get_account(state._overlay_tree, get_tree_key_for_version(addr))
+    if account is not None:
+        return account
+    
+    return trie_get(state._main_trie, address)
+
+def set_account(state: State, address: Address, account: Optional[Account]) -> None:
+    verkle_set_account(state._overlay_tree, get_tree_key_for_nonce(addr), account)
+
+def get_storage(state: State, address: Address, key: Bytes) -> U256:
+    value = state._overlay_tree.get(get_tree_key_for_storage_slot(addr, slot))
+    if value is not None:
+        return value
+        
+    trie = state._storage_tries.get(address)
+    if trie is None:
+        return U256(0)
+
+    value = trie_get(trie, key)
+
+    assert isinstance(value, U256)
+    return value
+
+def set_storage(
+    state: State, address: Address, key: Bytes, value: U256
+) -> None:
+    state._overlay_tree.set(get_tree_key_for_storage_slot(addr, slot), value)
+```
+
+## Rationale
+
+This approach doesn't convert the state, which is left to a subsequent EIP. This is meant as a stopgap in case we decide to push the conversion itself to a later time. It has the advantage of simplicity, which means that the verge fork could happen at the same time as other, simpler EIPs. It also requires no change at the consensus layer.
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Test Cases
+
+<!--
+  This section is optional for non-Core EIPs.
+
+  The Test Cases section should include expected input/output pairs, but may include a succinct set of executable tests. It should not include project build files. No new requirements may be be introduced here (meaning an implementation following only the Specification section should pass all tests here.)
+  If the test suite is too large to reasonably be included inline, then consider adding it as one or more files in `../assets/eip-####/`. External links will not be allowed
+
+  TODO: Remove this comment before submitting
+-->
+
+## Reference Implementation
+
+ * `transition-post-genesis` branch in `github.com/gballet/go-ethereum` implements this when setting `--override.overlay-stride=0` on the command line.
```

Comment:
Avoid linking to external sources. See EIP-1 for a list of permitted external resources. The "Reference Implementation" section is optional, so you can just leave it out. The other option is to put it in the `assets` directory.